src/re.rs

```diff
@@ -829,6 +837,26 @@ impl<'t> Iterator for SubCapturesPos<'t> {
     }
 }
 
+/// An Iterator over named capture groups as a tuple with the group
+/// name and the value.
+///
+/// `'t` is the lifetime of the matched text.
+pub struct SubCapturesNamed<'t>{
+    caps: &'t Captures<'t>,
+    inner: Option<Iter<'t, String, usize>>,
+}
+
+impl<'t> Iterator for SubCapturesNamed<'t> {
+    type Item = (&'t str, &'t str);
+
+    fn next(&mut self) -> Option<(&'t str, &'t str)> {
+        match self.inner.as_mut().map(|it| it.next()).unwrap_or(None) {
+            Some((name, pos)) => Some((name, self.caps.at(*pos).unwrap_or(""))),
```

Comment:
I didn't catch this the first time around, but given my comments in https://github.com/rust-lang/regex/issues/46#issuecomment-76642886, I think the iterator should probably return `Option<Option<(&'t str, &'t str)>>`. i.e., remove `unwrap_or("")`. Adding the `unwrap_or` in the iterator prevents the client from being able to differentiate between an empty matching capture group and a non-matching capture group.

This only slightly impacts the ergonomics of the iterator. e.g., you can do `caps.iter_name().filter_map(|x|x)` to skip non-matching captures of `caps.iter_named().map(|x| x.unwrap_or(""))` to get the behavior you have here.